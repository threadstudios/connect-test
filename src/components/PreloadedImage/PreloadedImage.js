import Inferno from 'inferno';
import { compose, branch, withLifecycle, withState, renderComponent, defaultProps } from 'incompose';
import Loading from '../Loading/Loading';

const PreloadedImage = ({url, className, loaded}) => {
    return (
        <div className={[className, loaded ? `${className}--loaded` : ''].join(' ')} style={{backgroundImage: `url(${url})` }}></div>
    );
}

export default compose(
    withState('loaded', 'setLoaded', false),
    branch(
        (props) => !props.loaded,
        withLifecycle({
            componentDidMount() {
                const img = new Image();
                img.src = this.props.url;
                img.onload = () => {
                    this.props.setLoaded(true);
                }
            }
        })
    ),
    branch(
        (props) => !props.loaded,
        renderComponent(Loading)
    )
)(PreloadedImage);