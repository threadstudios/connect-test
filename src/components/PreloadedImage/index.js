import PreloadedImage from './PreloadedImage';

export default PreloadedImage;