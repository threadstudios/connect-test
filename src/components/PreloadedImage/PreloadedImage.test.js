import Inferno from 'inferno';
import { renderToSnapshot, renderIntoDocument, findRenderedDOMElementWithClass, findRenderedVNodeWithType } from 'inferno-test-utils';
import PreloadedImage from './PreloadedImage';

describe("A Preloaded Image Component", () => {

    it("should render without an error", () => {
        expect(renderToSnapshot(<PreloadedImage />)).toMatchSnapshot();
    })

    it("should display a loader before the image is displayed", () => {
        const Preloader = renderIntoDocument(<PreloadedImage url="https://www.jaguar.co.uk/Images/X540_72frame_turn_V03_Desktop_02_tcm91-387710_desktop_1366x768.jpg?v=2" className="foo" />);
        const result = findRenderedDOMElementWithClass(Preloader, "loading");
        expect(result).toBeDefined();
    })

    it("should display the image given it has loaded", () => {

        const wrapper = renderIntoDocument(<PreloadedImage 
        url="https://www.jaguar.co.uk/Images/X540_72frame_turn_V03_Desktop_01_tcm91-387710_desktop_1366x768.jpg?v=2" 
        className="foo"
         />);

        const preloader = findRenderedVNodeWithType(wrapper, PreloadedImage);
        preloader.children.updateStateValue(true);

        const imageContainer = findRenderedDOMElementWithClass(wrapper, 'foo--loaded');

        expect(imageContainer).toBeDefined();
        expect(renderToSnapshot(preloader)).toMatchSnapshot();

    })

})