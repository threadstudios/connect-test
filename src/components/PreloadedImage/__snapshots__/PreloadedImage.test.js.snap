// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`A Preloaded Image Component should display the image given it has loaded 1`] = `
<div
  className="foo"
  url="https://www.jaguar.co.uk/Images/X540_72frame_turn_V03_Desktop_01_tcm91-387710_desktop_1366x768.jpg?v=2"
/>
`;

exports[`A Preloaded Image Component should render without an error 1`] = `<div />`;
